lib/crossdoc/tree.rb

```diff
@@ -36,11 +36,31 @@ def download
         elsif @src.index('./')==0
           @io = open(@src.gsub('./', Dir.pwd + '/'))
         else # assume it's a URL
-          @io = URI.open(@src)
+          @io = URI.open @src
+          if @is_svg || num_images < 6
+            return
+          end
+          img = MiniMagick::Image.open @src
```

Comment:
Can we open this image in MiniMagick from the io object that we already have? Seems a bit wasteful to open a remote url twice. If we can't directly open it from io, maybe save the io to a temp file and open it in MiniMagick from there.

Reply:
![CleanShot 2024-04-12 at 13 35 08](https://github.com/TinyMission/crossdoc/assets/25541844/54bf4300-cda1-4249-b8af-92448a301ddf)

There needs to be a separate open to avoid modifying the user's original file, if I'm understanding the documentation correctly. There doesn't seem to be a way to "tap into" the existing file descriptor. I think it's worthwhile to keep the original, in case this processing gets something wrong.

> maybe save the io to a temp file and open it in MiniMagick from there.
Doesn't this still result in the same number of opens? Or does it improve performance and/or efficiency in a way I'm not seeing?

Reply:
In the case where the url is a remote URL on a separate server, `URI.open` will download the image, then `MiniMagick::Image.open` will download it again. If we could download it once and then open the IO object and the MiniMagick object from the same location, it would skip the second download (although it will still open the file twice, and I guess MiniMagick will create a separate temp file for the working copy)